### PR TITLE
x86: Support x2AVIC on AMD Genoa

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 1305

AMD support one feature x2AVIC on Genoa.

### CODE REVIEW

**Comment by nanliu-r** [comment]
Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.x2avic_test.q35: PASS (157.38 s)

@PaulYuuu Could you help to review this? thanks.

**Comment by nanliu-r** [comment]
Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.x2avic_test.q35: PASS (108.87 s)

File: qemu/tests/x2avic_test.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    try:
+        session = vm.wait_for_login(timeout=timeout)
+        if params.get("os_type") == "linux":
+            output = session.cmd_output('dmesg | grep x2apic')
+            if "x2apic enabled" not in output:
+                if "x2apic: enabled" not in output:
+                    test.fail("x2apic is not enabled inside guest.")
+    finally:
+        session.close()
+    kvm_stat_output = str(process.system_output("kvm_stat -1"))
```
Comment (PaulYuuu) [inline qemu/tests/x2avic_test.py:37]:
```suggestion
    kvm_stat_output = process.getoutput("kvm_stat -1")
```

File: qemu/tests/x2avic_test.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    flags = params["flags"]
+    check_cpu_flags(params, flags, test)
+
+    params["start_vm"] = "yes"
+    vm = env.get_vm(params["main_vm"])
+    env_process.preprocess_vm(test, params, env, vm.name)
```
Comment (PaulYuuu) [inline qemu/tests/x2avic_test.py:25]: What resources need to be prepared here?

File: qemu/tests/x2avic_test.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    timeout = float(params.get("login_timeout", 240))
```
Comment (PaulYuuu) [inline qemu/tests/x2avic_test.py:26]: Default timeout is 240.

File: qemu/tests/x2avic_test.py (hunk 1 above)
Comment (nanliu-r) [reply qemu/tests/x2avic_test.py]: > 

Update and thanks.

File: qemu/tests/x2avic_test.py (hunk 2 above)
Comment (nanliu-r) [reply qemu/tests/x2avic_test.py]: > What resources need to be prepared here?

updated to `vm.create()`, Thanks.

File: qemu/tests/x2avic_test.py (hunk 3 above)
Comment (nanliu-r) [reply qemu/tests/x2avic_test.py]: > Default timeout is 240.

Updated and thanks.

**Comment by nanliu-r** [comment]
Test results.
 (1/2) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.x2avic_test.q35: STARTED
 (1/2) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.x2avic_test.q35: PASS (125.74 s)
 (2/2) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.x2avic_test.q35: STARTED
 (2/2) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.x2avic_test.q35: PASS (136.58 s)

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**